site/components/src/main/java/com/visitscotland/brxm/utils/SiteProperties.java

```
@@ -157,7 +157,7 @@ public Boolean getFormsMarketoIsProduction() {
     }
 
     public String getSiteId() {
-        return readString(SITE_ID);
+        return readOptionalString(SITE_ID).orElse("");
```

Again, just feels a bit backwards to return an empty string in the event that the Site ID does not exist, do we need a fallback or to throw an exception?

---

In this case the fallback is an empty String. 

When you read a property with `readString` and that property does not contain a value (or an accepted value) the property logs a warning, which should be reviewed by a developer. However readOptionalString notes that it is not unexpected that the property doesn't exist.

Now, you have a point there and what I should have done is to propagate the Optional<String> instead of returning a default value of `""`. I probably didn't do that to minimize the scope of this PR